### PR TITLE
Simplify logging package for greater flexibility.

### PR DESCRIPTION
**Pull request by jmalloc**
This package removes the `Logger` type from the logging package, leaving only the icons and some functions for formatting "standard" log messages.

This is less prescriptive, leaving the engine to make its own decisions about how logging is performed, but still provided some useful utilities.

### CODE REVIEW

File: logging/icon.go [hunk 1]
```diff
@@ -1,87 +1,155 @@
...
-	SystemIcon = "⚙"
+	SystemIcon Icon = "⚙"
 
 	// SeparatorIcon is an icon used to separate strings of unrelated text inside a
 	// log message. It is a large bullet, intended to have a large visual impact.
-	SeparatorIcon = "●"
+	SeparatorIcon Icon = "●"
 )
 
+// Icon athe type for strings that contain icons.
```
Comment (koden-km) [inline logging/icon.go]: _athe_ ?

Comment (koden-km) [reply logging/icon.go]: Also, I wonder if this Icon type definition should be done before the const group above? First thing I did when seeing it in the const group was to go find it and see what it was.

File: logging/icon.go [hunk 2]
```diff
@@ -1,87 +1,155 @@
...
+	s := i.String()
+	if i == "" {
+		s = " "
+	}
+
+	n, err := io.WriteString(w, s)
+	return int64(n), err
+}
+
+// WithLabel returns a log component that renders the icon with a label.
```
Comment (koden-km) [inline logging/icon.go]: The `IconWithLabel` type is not referred to as a _a log component_ in its doc block? Just confirming that this is still the intended doc comment for the return type here.

File: logging/icon.go [hunk 3]
```diff
@@ -1,87 +1,155 @@
...
+}
+
+// IconWithLabel is a container for an icon and its associated text label.
+type IconWithLabel struct {
+	Icon  Icon
+	Label string
+}
+
+func (i IconWithLabel) String() string {
+	return i.Icon.String() + " " + i.Label
```
Comment (koden-km) [inline logging/icon.go]: _[Nitpick]_ Do Go string concatenations with `+` cause garbage? If so, would a fmt.Sprintf() make less string garbage?

Comment (jmalloc) [reply logging/icon.go]: A `sprintf()` will "make more garbage" because it has to box values to pass them as `interface{}`.

**Pull request merged**